DLL loading fails when using pyenv and poetry on Windows

When using pyenv for python versions, and poetry for virtual environments, the following bit of [lib loading code](https://github.com/dmlc/treelite/blob/cffe8943e3cb9e1eca609d38b147e8bfeb6ceadb/python/treelite/core.py#L34-L38) for treelite breaks with the error:

```bash
ImportError while loading conftest 'C:\Users\daarashaw\windows-dev\my_repo\tests\conftest.py'.
...
..\..\AppData\Local\pypoetry\Cache\virtualenvs\my-repo-py3.10\lib\site-packages\tl2cgen\__init__.py:6: in <module>
    from .core import _dump_compiler_ast, _py_version, annotate_branch, generate_c_code
..\..\AppData\Local\pypoetry\Cache\virtualenvs\my-repo-py3.10\lib\site-packages\tl2cgen\core.py:8: in <module>
    import treelite
..\..\AppData\Local\pypoetry\Cache\virtualenvs\my-repo-py3.10\lib\site-packages\treelite\__init__.py:5: in <module>
    from . import frontend, gtil, model_builder, sklearn
..\..\AppData\Local\pypoetry\Cache\virtualenvs\my-repo-py3.10\lib\site-packages\treelite\frontend.py:10: in <module>
    from . import compat
..\..\AppData\Local\pypoetry\Cache\virtualenvs\my-repo-py3.10\lib\site-packages\treelite\compat.py:9: in <module>
    from .core import _LIB, TreeliteError, _check_call
..\..\AppData\Local\pypoetry\Cache\virtualenvs\my-repo-py3.10\lib\site-packages\treelite\core.py:58: in <module>
    _LIB = _load_lib()
..\..\AppData\Local\pypoetry\Cache\virtualenvs\my-repo-py3.10\lib\site-packages\treelite\core.py:36: in _load_lib
    os.add_dll_directory(
..\..\.pyenv\pyenv-win\versions\3.10.11\lib\os.py:1118: in add_dll_directory
    cookie = nt._add_dll_directory(path)
E   FileNotFoundError: [WinError 3] The system cannot find the path specified: 'C:\\Users\\daarashaw\\.pyenv\\pyenv-win\\versions\\3.10.11\\Library\\bin'
```
### poetry env info
```bash
Virtualenv
Python:         3.10.11
Implementation: CPython
Path:           C:\Users\daarashaw\AppData\Local\pypoetry\Cache\virtualenvs\my-repo-py3.10
Executable:     C:\Users\daarashaw\AppData\Local\pypoetry\Cache\virtualenvs\my-repo-py3.10\Scripts\python.exe
Valid:          True

Base
Platform:   win32
OS:         nt
Python:     3.10.11
Path:       C:\Users\daarashaw\.pyenv\pyenv-win\versions\3.10.11
Executable: C:\Users\daarashaw\.pyenv\pyenv-win\versions\3.10.11\python.exe
```

In [here](https://github.com/dmlc/treelite/blob/cffe8943e3cb9e1eca609d38b147e8bfeb6ceadb/python/treelite/core.py#L34-L38), it doesn't check whether the folder exists, causing it to break. 

I've been able to manually solve it my making the `Library/bin` directory under my base virtualenv.

---
What's the purpose of this dll location check, is it for a specific python install method (eg via conda?)

I can make a quick PR fix to solve it later today. Should just need checking whether the folder exists before trying to add a dll from it.

--- a/python/treelite/core.py
+++ b/python/treelite/core.py
@@ -33,9 +33,10 @@ def _load_lib():
         return None  # type: ignore
     if sys.version_info >= (3, 8) and sys.platform == "win32":
         # pylint: disable=no-member
-        os.add_dll_directory(
-            os.path.join(os.path.normpath(sys.base_prefix), "Library", "bin")
-        )
+        lib_bin_path = os.path.join(os.path.normpath(sys.base_prefix), "Library", "bin")
+        if os.path.isdir(lib_bin_path):
+            os.add_dll_directory(lib_bin_path)
+
     lib = ctypes.cdll.LoadLibrary(lib_path[0])
     lib.TreeliteGetLastError.restype = ctypes.c_char_p
     lib.log_callback = _log_callback

